### ISSUE TITLE
Mac OSX builds not updating

### ISSUE THREAD

**Issue by hroest** [OpenMS/OpenMS#1546]
something is wrong with git on the mac osx in our cdash, they have not been updated since May and are now more than 200 commits behind: http://cdash.openms.de/viewUpdate.php?buildid=138082 

```
Update command failed: "/usr/local/bin/git" "fetch"
```

**Comment by jpfeuffer** [comment]
I looked into the problem. Git works fine again.
Still there are problems with the GCC builds after an update of homebrew, upgrading gcc to gcc5.
It seems like even the GCC builds use the clang-built contribs. The weird thing is, that I don't remember changing this, so it seems like this has been the case for a while.
Either the compilers became incompatible over the time, or there was really I change I cant explain. Did we build for gcc on Mac "recently"?

**Comment by hroest** [comment]
I think in the past this was not an issue but with  the upgrade to gcc 5 this may actually be the problem:

http://lists.alioth.debian.org/pipermail/debian-med-packaging/2015-July/032800.html

it thus seems that boost needs to be rebuilt, at least some other people also noticed this
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=790351

**Comment by jpfeuffer** [comment]
Ah, thanks for the links! Matches the errors in the log, that only concern boost.

**Comment by aiche** [comment]
I updated the scripts on the build machine to select an appropriate contrib, lets see if this fixes the gcc5 issues.

**Comment by hroest** [comment]
something broke again for pyopenms, the same revision bec4d128ab34934cd8d3d270011ab932c5bd8c6a produced an complete build on Wednesday but then failed on Thursday: 

- http://cdash.openms.de/buildSummary.php?buildid=139421
- http://cdash.openms.de/buildSummary.php?buildid=139520

it seems some update was installed in the meantime as for example the version of doxygen is different

**Comment by aiche** [comment]
I tried directly on the machine and produced the following error 

```
[100%] Built target pyopenms_create_cpp
Extracting in /var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN
Now working in /var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34
Building a Distribute egg in /Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS
Traceback (most recent call last):
  File "setup.py", line 248, in <module>
    scripts = scripts,
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/setuptools/dist.py", line 225, in __init__
    _Distribution.__init__(self,attrs)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 287, in __init__
    self.finalize_options()
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/setuptools/dist.py", line 257, in finalize_options
    ep.require(installer=self.fetch_build_egg)
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/pkg_resources.py", line 2025, in require
    working_set.resolve(self.dist.requires(self.extras),env,installer))
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/pkg_resources.py", line 2235, in requires
    dm = self._dep_map
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/pkg_resources.py", line 2464, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/pkg_resources.py", line 2497, in _compute_dependencies
    common = frozenset(reqs_for_extra(None))
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/pkg_resources.py", line 2494, in reqs_for_extra
    if req.marker_fn(override={'extra':extra}):
  File "/private/var/folders/s2/sj5gxp_n4cx41k5jfbfflklc0000gn/T/tmp5KxYWN/distribute-0.6.34/_markerlib/markers.py", line 109, in marker_fn
    return eval(compiled_marker, environment)
  File "<environment marker>", line 1, in <module>
NameError: name 'sys_platform' is not defined
/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/distribute-0.6.34-py2.7.egg
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    distribute_setup.use_setuptools()
  File "/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/distribute_setup.py", line 152, in use_setuptools
    return _do_download(version, download_base, to_dir, download_delay)
  File "/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/distribute_setup.py", line 132, in _do_download
    _build_egg(egg, tarball, to_dir)
  File "/Users/aiche/NIGHTLY/builds/mac-osx-10.8-clang-5.0-LLVM3.3svn-Unix_Makefiles-Release/pyOpenMS/distribute_setup.py", line 123, in _build_egg
    raise IOError('Could not build the egg.')
IOError: Could not build the egg.
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms] Error 1
make[2]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/all] Error 2
make[1]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/rule] Error 2
make: *** [pyopenms] Error 2
```

Since I'm not a python expert, @hroest any ideas?

**Comment by jpfeuffer** [comment]
What about the "errors" from QcML? I don't get why they are produced. Are they real errors?

@aiche: Did you do upgrades on the machine? seems like whole cmake was upgraded.

**Comment by aiche** [comment]
> What about the "errors" from QcML? I don't get why they are produced. Are they real errors?

Those are false positives from the ctest parser. I'm still working on making cdash ignore them.

> Did you do upgrades on the machine? seems like whole cmake was upgraded.

I did a `brew upgrade` which included python as well as cmake, but those changes shouldn't lead to such errors. I found some results for `NameError: name 'sys_platform' is not defined` but none of the propose solutions worked.

**Comment by hroest** [comment]
@aiche not sure what is happening here, it seems to be some problem with distribute. can you try to remove and re-install distribute, e.g.

`pip uninstall distribute && pip install distribute`

and try again? The newest version is 0.7.3 maybe that already solves the problem? 

@uweschmitt do you have any idea what is going wrong here?

**Comment by aiche** [comment]
Reinstalled `distribute` and `setuptools`, but it didn't solve the issue. I still get the above error.

**Comment by hroest** [comment]
hmm I dont really have an idea what `distribute_setup.py` is trying to accomplish here but it seems to be some outdated version of how to install distribute. I think @uweschmitt set it up at the time, maybe he has some insights into how to improve the situation?

**Comment by hroest** [comment]
@aiche @jpfeuffer I have now run into the same issues on another machine here (Linux) and I managed to solve the issue (I got the exact same error as above)

Apparently, distribute is not really supported any more and now part of setuptools:
http://stackoverflow.com/questions/6344076/differences-between-distribute-distutils-setuptools-and-distutils2

I merged an older PR and fixed the merge conflict in #1717 and I am currently compiling pyOpenMS on my machine ...

**Comment by timosachsenberg** [comment]
can we close this issue?
